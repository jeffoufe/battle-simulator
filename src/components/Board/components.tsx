import styled from 'styled-components';
import { breakpoints } from '../../constants';

export const BoardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: calc(100vh - 60px);

  @media only screen and (min-width: ${breakpoints.tablet}) {
    flex-direction: row;
    align-items: center;

    > div {
        width: calc(100% / 3);
    }
  }
`

BoardContainer.displayName = 'BoardContainer';