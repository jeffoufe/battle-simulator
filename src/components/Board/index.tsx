import React from 'react';
import { BoardContainer } from './components';
import { Title, Character, Actions } from '..';
import WARRIOR_IMAGE from '../../assets/warrior.png';
import MONSTER_IMAGE from '../../assets/monster.png';

const Board = () => (
  <>
    <Title>Battle Simulator</Title>
    <BoardContainer>
      <Character
        image={WARRIOR_IMAGE}
        order={0}
      />
      <Actions />
      <Character
        image={MONSTER_IMAGE}
        order={1}
      />
    </BoardContainer>
  </>
)

export default Board;
