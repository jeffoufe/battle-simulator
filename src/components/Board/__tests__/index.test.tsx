import React from 'react';
import { shallow } from 'enzyme';
import Board from '..'

describe('Board Components', () => {
    it('BoardContainer', () => {
        const wrapper = shallow(
            <Board />
        );
        expect(wrapper).toMatchSnapshot();
    })
})