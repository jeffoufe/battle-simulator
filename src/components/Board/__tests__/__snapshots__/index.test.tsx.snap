// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Components BoardContainer 1`] = `
<Fragment>
  <Title>
    Battle Simulator
  </Title>
  <BoardContainer>
    <Character
      image="warrior.png"
      order={0}
    />
    <Actions />
    <Character
      image="monster.png"
      order={1}
    />
  </BoardContainer>
</Fragment>
`;
